compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -8582,7 +8584,25 @@ public void visit(BLangReDisjunction reDisjunction) {
 
     @Override
     public void visit(BLangReSequence reSequence) {
-        reSequence.termList.forEach(this::rewriteExpr);
+        // if the expr is only `[]` replace it with `(?:)` 
+        boolean desugared = false;
+        if (reSequence.termList.size() == 1 
+                && reSequence.termList.get(0).getKind() == NodeKind.REG_EXP_ATOM_QUANTIFIER) {
+            BLangReAtomQuantifier reAtomQuantifier = ((BLangReAtomQuantifier) reSequence.termList.get(0));
+            if (reAtomQuantifier.atom.getKind() == NodeKind.REG_EXP_CHARACTER_CLASS 
+                    && ((BLangReCharacterClass) ((BLangReCharacterClass) reAtomQuantifier.atom)).charSet == null) {
+                    desugared = true;
+                    reAtomQuantifier.atom = createLiteral(
+                            reAtomQuantifier.pos, symTable.stringType, RE_CAPTURING_GROUP_PATTERN);
+                    if (reAtomQuantifier.quantifier == null) {
+                        reAtomQuantifier.quantifier = ASTBuilderUtil.createEmptyQuantifier(
+                                reAtomQuantifier.pos, symTable.anydataType, symTable.stringType);
+                }
+            }
+        }
+        if (!desugared) {
+            reSequence.termList.forEach(this::rewriteExpr);
+        }
         result = reSequence;
```

Comment:
I think it will be more readable if we can change the whole method like this
```java
        if (reSequence.termList.size() == 1
                && reSequence.termList.get(0).getKind() == NodeKind.REG_EXP_ATOM_QUANTIFIER) {
            BLangReAtomQuantifier reAtomQuantifier = ((BLangReAtomQuantifier) reSequence.termList.get(0));
            if (reAtomQuantifier.atom.getKind() == NodeKind.REG_EXP_CHARACTER_CLASS
                    && (((BLangReCharacterClass) reAtomQuantifier.atom)).charSet == null) {
                reAtomQuantifier.atom = createLiteral(
                        reAtomQuantifier.pos, symTable.stringType, RE_CAPTURING_GROUP_PATTERN);
                if (reAtomQuantifier.quantifier == null) {
                    reAtomQuantifier.quantifier = ASTBuilderUtil.createEmptyQuantifier(
                            reAtomQuantifier.pos, symTable.anydataType, symTable.stringType);
                }
                result = reSequence;
                return;
            }
        }
        reSequence.termList.forEach(this::rewriteExpr);
        result = reSequence;
```